### PR TITLE
chore: update release manifests and examples with new spec

### PR DESCRIPTION
**Pull request by whynowy**

### CODE REVIEW

File: manifests/cluster-install/kustomization.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
-bases:
+resources:
```
Comment (whynowy) [inline manifests/cluster-install/kustomization.yaml:4]: `bases` is deprecated.

File: manifests/namespace-install/kustomization.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
-bases:
+resources:
```
Comment (whynowy) [inline manifests/namespace-install/kustomization.yaml:4]: `bases` is deprecated.

File: manifests/base/kustomization.yaml [hunk 3]
```diff
@@ -1,16 +1,48 @@
...
+  - name: argoproj/sensor-controller
+    newTag: v0.14.0
+
+patchesStrategicMerge:
+  - |-
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: gateway-controller
+    spec:
```
Comment (whynowy) [inline manifests/base/kustomization.yaml:26]: Put all the `version` patch in one file to make it easy to update.

File: manifests/namespace-install.yaml [hunk 4]
```diff
@@ -171,6 +175,12 @@ spec:
               fieldPath: metadata.namespace
         - name: CONTROLLER_CONFIG_MAP
           value: gateway-controller-configmap
+        - name: GATEWAY_CLIENT_IMAGE
```
Comment (VaibhavPage) [inline manifests/namespace-install.yaml:178]: lets get rid of CLIENT_IMAGE env var.

Comment (whynowy) [reply manifests/namespace-install.yaml:178]: Can you give more detail? @VaibhavPage

**Pull request merged**